litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go

```diff
@@ -123,16 +123,21 @@ func WorkFlowRunHandler(input model.WorkflowRunInput, r store.StateData) (string
 	}
 
 	//err = database.UpsertWorkflowRun(database.WorkflowRun(newWorkflowRun))
-	err = database.UpsertWorkflowRun(input.WorkflowID, database.WorkflowRun{
+	count, err := database.UpsertWorkflowRun(input.WorkflowID, database.WorkflowRun{
 		WorkflowRunID: input.WorkflowRunID,
 		LastUpdated:   strconv.FormatInt(time.Now().Unix(), 10),
 		ExecutionData: input.ExecutionData,
+		Completed:     input.Completed,
 	})
 	if err != nil {
 		log.Print("ERROR", err)
 		return "", err
 	}
 
+	if count == 0 {
+		return "Workflow Run Accepted", nil
```

Comment:
Can you change the return message? The message is the same if count != 0

Reply:
Not needed, the subscriber as it is doesn't need to know about the duplicate events because it cannot do anything about these events. The events are generated by the workflow controller

Reply:
Updated